access.lua

```diff
@@ -247,66 +255,14 @@ end
 -- has to be sent, even if the user is already authenticated.
 --
 
-if conf["skipped_urls"] then
-    for _, url in ipairs(conf["skipped_urls"]) do
-        if (hlp.string.starts(ngx.var.host..ngx.var.uri..hlp.uri_args_string(), url)
-        or  hlp.string.starts(ngx.var.uri..hlp.uri_args_string(), url))
-        then
-            logger.debug("Skipping "..ngx.var.uri)
-            return hlp.pass()
-        end
-    end
-end
-
-if conf["skipped_regex"] then
-    for _, regex in ipairs(conf["skipped_regex"]) do
-        if (hlp.match(ngx.var.host..ngx.var.uri..hlp.uri_args_string(), regex)
-        or  hlp.match(ngx.var.uri..hlp.uri_args_string(), regex))
-        then
-            logger.debug("Skipping "..ngx.var.uri)
-            return hlp.pass()
-        end
-    end
-end
-
---
--- 5. Protected URLs
---
--- If the URL matches one of the `protected_urls` in the configuration file,
--- we have to protect it even if the URL is also set in the `unprotected_urls`.
--- It could be useful if you want to unprotect every URL except a few
--- particular ones.
---
-
-function is_protected()
-    if not conf["protected_urls"] then
-        conf["protected_urls"] = {}
-    end
-    if not conf["protected_regex"] then
-        conf["protected_regex"] = {}
-    end
-
-    for _, url in ipairs(conf["protected_urls"]) do
-        if hlp.string.starts(ngx.var.host..ngx.var.uri..hlp.uri_args_string(), url)
-        or hlp.string.starts(ngx.var.uri..hlp.uri_args_string(), url) then
-            logger.debug(ngx.var.uri.." is in protected_urls")
-            return true
-        end
-    end
-    for _, regex in ipairs(conf["protected_regex"]) do
-        if hlp.match(ngx.var.host..ngx.var.uri..hlp.uri_args_string(), regex)
-        or hlp.match(ngx.var.uri..hlp.uri_args_string(), regex) then
-            logger.debug(ngx.var.uri.." is in protected_regex")
-            return true
-        end
-    end
-
-    logger.debug(ngx.var.uri.." is not in protected_urls/regex")
-    return false
+if longest_skipped_match ~= ""
+and string.len(longest_skipped_match) >= string.len(longest_protected_match) then
```

Comment:
Shouldn't we add an other condition like that:
```lua
string.len(longest_skipped_match) > string.len(longest_protected_match)
```

Reply:
Why should we add this condition?

Reply:
Oh my bad, I wrote the bad test that I think...
It was this that I thought:
```lua
string.len(longest_skipped_match) > string.len(longest_unprotected_match)
```

Because we have 3 type of "url match". Here we check if the skipped url is the url which is the best match. To be sure about that we need to compare with the 2 other.

By example we have : 
- `longest_unprotected_match` : `/pathA/api`
- `longest_skipped_match` : `/pathA/`.

In this case the unprotected should be a priority.

Reply:
:100: